runtime/client_http_response_test.go

```diff
@@ -76,11 +77,13 @@ func TestReadAndUnmarshalNonStructBody(t *testing.T) {
 		time.Second,
 	)
 	ctx := context.Background()
+	ctx = runtime.WithLogger(ctx, bgateway.ActualGateway.Logger)
+
 	req := zanzibar.NewClientHTTPRequest(ctx, "bar", "echo", client)
 
 	err = req.WriteJSON("POST", baseURL+"/bar/echo", nil, myJson{})
 	assert.NoError(t, err)
-	res, err := req.Do(context.Background())
+	res, err := req.Do()
```

Comment:
I don't see that there is a difference for deadline propagation. It was redundant before since there were no mutations done to the context, but now `WriteJSON` modifies the context so it's actually confusing / misleading to pass context into `Do()`.